src/validation.cpp

```diff
@@ -3045,14 +3046,8 @@ bool CheckBlock(const CBlock& block, CValidationState& state, const Consensus::P
 
 bool IsWitnessEnabled(const CBlockIndex* pindexPrev, const Consensus::Params& params)
 {
-    LOCK(cs_main);
-    return (VersionBitsState(pindexPrev, params, Consensus::DEPLOYMENT_SEGWIT, versionbitscache) == ThresholdState::ACTIVE);
-}
-
-bool IsNullDummyEnabled(const CBlockIndex* pindexPrev, const Consensus::Params& params)
-{
-    LOCK(cs_main);
-    return (VersionBitsState(pindexPrev, params, Consensus::DEPLOYMENT_SEGWIT, versionbitscache) == ThresholdState::ACTIVE);
+    int height = pindexPrev == nullptr ? 0 : pindexPrev->nHeight + 1;
```

Comment:
I don't think that works for `UpdateUncommittedBlockStructures` or `BlockAssembler::CreateNewBlock`, so you'd still need this function to be available.

Reply:
Right these functions test for the next block to be generated/submitted based on the parent,  for them you may pass `pindexPrev->nHeight + 1` to `IsWitnessEnabled`, should work but need to refactor also callers ?

Reply:
Right, I think this could also be refactored to take a `height` parameter and just return whether segwit should be enforced for a block at that height.

We can leave that for a follow-up PR if people think it's worthwhile.

Reply:
If it was going to take a height parameter, might as well just have the test in place directly. I don't think it's worth fiddling with for just segwit.